packages/utils/src/ts-eslint/Rule.ts

```diff
@@ -424,6 +425,18 @@ interface RuleListener {
   WithStatement?: RuleFunction<TSESTree.WithStatement>;
   YieldExpression?: RuleFunction<TSESTree.YieldExpression>;
 }
+type RuleListenerExitSelectors = {
+  [K in keyof RuleListenerBaseSelectors as `${K}:exit`]: RuleListenerBaseSelectors[K];
+};
+interface RuleListenerCatchAllBaseCase {
+  [nodeSelector: string]: RuleFunction | undefined;
+}
+// Interface to merge into for anyone that wants to add more selectors
+interface RuleListenerExtension {}
```

Comment:
That was just me adding some future-proofing in case anyone was doing something weird.
Actually don't need it - but I think it's fine to leave it undocumented for the moment.